internal/services/eventgrid/eventgrid_partner_namespace_channel_resource.go

```diff
@@ -0,0 +1,455 @@
+}
+
+// MessageForActivation is a problematic field as the API generates a custom default message that can be longer than the allowed length if not included.
+// As such it has been excluded and left to the server to set the default.
+// https://github.com/Azure/azure-rest-api-specs/issues/37280
+func (EventGridPartnerNamespaceChannelResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.StringMatch(
+				regexp.MustCompile("^[-a-zA-Z0-9]{3,50}$"),
+				"`name` must be between 3 and 50 characters. It can contain only letters, numbers and hyphens (-).",
+			),
+		},
+		"partner_namespace_id": commonschema.ResourceIDReferenceRequiredForceNew(&partnernamespaces.PartnerNamespaceId{}),
+		"partner_topic": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+						ForceNew: true,
+						ValidateFunc: validation.StringMatch(
+							regexp.MustCompile("^[-a-zA-Z0-9]{3,50}$"),
+							"`name` must be between 3 and 50 characters. It can contain only letters, numbers and hyphens (-).",
+						),
+					},
+					"subscription_id": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.IsUUID,
+					},
+					"resource_group_name": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"source": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ForceNew:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+					"event_type_definitions": {
+						Type:     pluginsdk.TypeList,
+						Optional: true,
+						MaxItems: 1,
+						Elem: &pluginsdk.Resource{
+							Schema: map[string]*pluginsdk.Schema{
+								"inline_event_type": {
+									Type:     pluginsdk.TypeSet,
+									Required: true,
+									Elem: &pluginsdk.Resource{
+										Schema: map[string]*pluginsdk.Schema{
+											"name": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ValidateFunc: validation.StringLenBetween(1, 128),
+											},
+											"display_name": {
+												Type:         pluginsdk.TypeString,
+												Required:     true,
+												ValidateFunc: validation.StringLenBetween(1, 128),
+											},
+											"data_schema_url": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ValidateFunc: validation.IsURLWithHTTPorHTTPS,
+											},
+											"description": {
+												Type:         pluginsdk.TypeString,
+												Optional:     true,
+												ValidateFunc: validation.StringLenBetween(0, 256),
```

Comment:
since `description` is optional, the `ValidateFunc` should be `validation.StringLenBetween(1, 256),`?